internal/server/middleware.go

```diff
@@ -182,3 +182,76 @@ func getCookie(req *http.Request, name string) (string, error) {
 	}
 	return url.QueryUnescape(cookie.Value)
 }
+
+func OrganizationFromDomain(defaultOrgName, defaultOrgDomain string) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		db := getDB(c)
+		host := c.Request.Header.Get("Host")
+
+		// remove port
+		if i := strings.Index(host, ":"); i >= 0 {
+			host = host[0:i]
+		}
+
+		if len(host) == 0 {
+			// tests are lazy and don't set Host
+			host = defaultOrgDomain
```

Comment:
Seems like this could be unexpected if the client making the API request doesn't know to set this header value. Should we fail without the host?

Reply:
this is set automatically by all clients to whatever the domain is, even if you don't know to set it.